room/compiler-processing/src/main/java/androidx/room/compiler/processing/javac/JavacAnnotation.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.room.compiler.processing.javac
+
+import androidx.room.compiler.processing.InternalXAnnotation
+import androidx.room.compiler.processing.XAnnotationBox
+import androidx.room.compiler.processing.XNullability
+import androidx.room.compiler.processing.XType
+import androidx.room.compiler.processing.XValueArgument
+import com.google.auto.common.AnnotationMirrors
+import javax.lang.model.element.AnnotationMirror
+
+internal class JavacAnnotation(
+    val env: JavacProcessingEnv,
+    val mirror: AnnotationMirror
+) : InternalXAnnotation {
+
+    override val simpleName: String
+        get() = mirror.annotationType.asElement().simpleName.toString()
+
+    override val type: XType by lazy {
+        JavacDeclaredType(env, mirror.annotationType, XNullability.NONNULL)
```

Comment:
nullability is UNKONWN here, no ?

Reply:
maybe I'm not fully understanding what this represents and you can better explain, but I was viewing it as an annotation on an annotated element is never null (ie if you get the list of annotations on an element none of those annotations are nullable). also, annotation parameters cannot be nullable so I don't see where a nullable annotation can come from

Reply:
oh then you are right, i wasn't thinking that / aware :/.